src/planner.cc

```diff
@@ -663,175 +668,205 @@ void Planner::Impl::bring_path_to_halt() {
   issue_motor_move_if_possible(true);
 }
 
+// Compute the backard pass of <segment_to_plan> given the
+// information (such as starting speed and defining axis) of the <next_segment>.
+// If <next_segment> is nullptr, the next segment is assumed to not exist (end
+// of the buffer). The function returns true if the final speed already connects
+// with the starting speed of the next segment and this is not the last segment.
+bool Planner::Impl::BackwardUpdateSegment(PlanningSegment *segment_to_plan,
+                                          const PlanningSegment *next_segment) {
+  // Handle special case of zero feedrate. The segment is dummy as any no amount
+  // of steps and be executed at zero speed.
+  if (segment_to_plan->target.speed == 0) {
+    return false;
+  }
+
+  const bool is_last_segment = next_segment == nullptr;
+  const enum GCodeParserAxis defining_axis =
+    segment_to_plan->target.defining_axis;
+  const AxisTarget &target = segment_to_plan->target;
+  PlannedProfile &planned = segment_to_plan->planned;
+
+  // Amount of absolute number of steps to be travelled in this segment
+  // by the defining axis.
+  const int delta_steps = abs(target.delta_steps[defining_axis]);
+  const GCodeParserAxis next_defining_axis =
+    is_last_segment ? defining_axis : next_segment->target.defining_axis;
+
+  // This speed is the starting speed of the defining axis of the next segment
+  // that we have to match our v2 with. The final speed we have to match is
+  // either zero if this is the last segment or v0 otherwise.
+  const double next_v0 =
+    is_last_segment
+      ? 0
+      : std::min(next_segment->planned.v1, next_segment->target.start_speed);
+
+  // We want to have continuous speeds. We calculate what would be that speed
+  // for the current defining axis.
+  const auto speed_factor =
+    get_speed_factor_for_axis(&target, next_defining_axis);
+
+  // Final speed constraint.
+  const double end_speed = (speed_factor != 0) ? next_v0 / speed_factor : 0;
+
+  // Check if the current segment v2 speed connects with the next
+  // (except if this is the very last segment).
+  // If we do, then we stop here. No need to compute the accel. It will happen
+  // in the forward pass. The new profile deceleration and travel will always
+  // be above or equal to the current profile. This will only then happen if
+  // the new backward profile v2 equals the previous planned profile v2.
+  if (planned.v2 == end_speed && !is_last_segment) {
```

Comment:
Sometimes comparing _exactly_ with float numbers is doable in particular if we have assigned things from the same calculation, but I think it makes sense to comment _why_ comparing two floats exactly is safe here
(as opposed to comparing that it is in very close proximity or some relative e.g. `<=` operator).

Reply:
(solved)